test/jit/test_with.py

```diff
@@ -548,3 +548,18 @@ def test_exit_incorrect_types(x, c):
             self.checkScript(
                 test_exit_incorrect_types, (test_tensor, ExitIncorrectTypes())
             )
+
+    def test_with_no_grad(self):
+        """
+        Check that torch.no_grad() works.
+        """
+        def test_fn(x):
+            # type: (Tensor) -> (Tuple[Tensor, bool])
+            with torch.no_grad():
+                y = x * x
+
+            return y, y.requires_grad
+
+        test_input = torch.randn(5)
```

Comment:
doesn't this test pass even if you don't have a no_grad block?

Reply:
I don't know what the scope of support for autograd generally is with torchscript, but I think you'd want a subset of the no_grad tests in test_autograd, i.e.:
https://github.com/pytorch/pytorch/blob/34025eb8265b03ef83256a1a9c42f8876c9428f5/test/test_autograd.py#L912-L973
https://github.com/pytorch/pytorch/blob/34025eb8265b03ef83256a1a9c42f8876c9428f5/test/test_autograd.py#L2026-L2059
https://github.com/pytorch/pytorch/blob/34025eb8265b03ef83256a1a9c42f8876c9428f5/test/test_autograd.py#L4210-L4234

Reply:
Cool, I adapted a handful of these for JIT. I'll consult my JIT reviewers on whether what I added is enough.